org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/matching/MatchLocator.java

```diff
@@ -1273,6 +1277,15 @@ private boolean skipMatch(JavaProject javaProject, PossibleMatch possibleMatch)
 	return false;
 }
 protected void locateMatches(JavaProject javaProject, PossibleMatch[] possibleMatches, int start, int length) throws CoreException {
+	IJavaSearchDelegate delegate = JavaSearchDelegateDiscovery.getInstance();
+	if( delegate != null ) {
+		delegate.locateMatches(this, javaProject, possibleMatches, start, length);
+	} else {
+		locateMatchesDefaultImpl(javaProject, possibleMatches, start, length);
```

Comment:
I guess the only question I have about this (and maybe @datho7561 can comment) is that you've opted to make the "default" behaviour a separate method call independent of the `IJavaSearchDelegate`. In previous usages of this approach (eg. for completion), the default approach was made to implement the `IJavaSearchDelegate` method and returns as a functional interface in the default case. Any reason that approach is not wanted here ? Is it a pain ?

Reply:
I agree that the default implementation should follow the interface if possible. I don't know the implementation details of the default implementation, but I think it should be possible.

Reply:
David pointed out it's not as clean beacuse the MatchLocator is also the parameter, so I guess the current approach would be fine.